spec/core/data_structures.md

```diff
@@ -4,195 +4,243 @@ Here we describe the data structures in the Tendermint blockchain and the rules
-- `Header`
-- `Version`
-- `BlockID`
-- `Time`
-- `Data` (for transactions)
-- `Commit` and `Vote`
-- `EvidenceData` and `Evidence`
+- [`Block`](#block)
+- [`Header`](#header)
+- [`Version`](#version)
+- [`BlockID`](#blockid)
+- [`Time`](#time)
+- [`Data` (for transactions)](#data)
+- [`Commit`](#commit)
+- [`Vote`](#vote)
+- [`EvidenceData`](#evidence_data)
+- [`Evidence`](#evidence)
 
 ## Block
 
 A block consists of a header, transactions, votes (the commit),
 and a list of evidence of malfeasance (ie. signing conflicting votes).
 
-```go
-type Block struct {
-    Header      Header
-    Txs         Data
-    Evidence    EvidenceData
-    LastCommit  Commit
-}
-```
+| Name       | Type                           | Description                                                                                                                                                                          | Validation                                                                                       |
+|------------|--------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
+| Header     | [Header](#header)              | Header corresponding to the block. This field contains information used throughout consensus and other areas of the protocol. To find out what it contains, visit [header] (#header) | Must adhere to the validation rules of [header](#header)                                         |
+| Data       | [Data](#data)                  | Data contains a list of transactions. The contents of the transaction is unknown to Tendermint.                                                                                      | This field can be empty or populated, but the data is unknown to Tendermint                      |
+| Evidence   | [EvidenceData](#evidence_data) | Evidence contains a list of infractions committed by validators.                                                                                                                     | Can be empty, but when populated the validations rules from [evidenceData](#evidence_data) apply |
+| LastCommit | [Commit](#commit)              | `LastCommit` includes one vote for every current validator.  All votes must either be for the previous block, nil or absent. All votes must have a valid signature from the corresponding validator.  The sum total of the voting power of the validators that voted must be greater than 2/3 of the total voting power of the complete validator set. The number of votes in a commit is limited to 10000 (see `types.MaxVotesCount`).                                                                                             | Can be empty for the initial height and must adhere to the validation rules of [commit](#commit).  |
```

Comment:
Doesn't a commit have to be empty for the initial height? Consider changing "Can" to "Must"